workflow-to-js-transformer/src/test/java/com/github/cafdataprocessing/workflow/transform/WorkflowJavaScriptExecutionTest.java

```diff
@@ -533,6 +533,31 @@ public void projectIdCustomDataTest()
 
     }
 
+    @Test(expectedExceptions = NullPointerException.class, description = "Tests that when custom data in workflow XML specifies a source "
+        + "of projectId and the projectId is Null, an Exception is thrown")
+    public void nullProjectIdCustomDataTest()
+            throws WorkerException, IOException, ScriptException, WorkflowTransformerException,
+            URISyntaxException, NoSuchMethodException, DataStoreException {
+        final String workflowJSStr = getWorkflowJavaScriptFromXML("/test_workflow_4.xml", null);
+        final Invocable invocable = getInvocableWorkflowJavaScriptFromJS(workflowJSStr);
+        final TestServices testServices = TestServices.createDefault();
```

Comment:
This test throws the exception before this line doesn't it? The rest of the test code should never be hit so I think it should be removed for clarity.